### PR TITLE
feat(Bilibilitv): support web multiLanguage & more

### PR DESCRIPTION
**Pull request by Faelayis**
**Describe the changes in this pull request:**

- Add cover ,multiLanguage
- Support all language in web (Previously english support only)

Add rich presence assets I need to change new clientId
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>

image
image
image
image
image
image

</details>

### CODE REVIEW

File: websites/B/bilibilitv/presence.ts [hunk 1]
```diff
@@ -1,97 +1,164 @@
...
-		presenceData.smallImageText = "Watching a show";
-		presenceData.details = getElement("h1.video-info__title");
-		presenceData.state = `Episode ${getEpisode("title")}`;
-		presenceData.startTimestamp = browsingStamp;
-		presenceData.buttons = [
-			{
-				label: "Watch this show",
-				url: `https://www.bilibili.tv/en/play/${urlpath[3]}`
+	if (document.location.hostname === "www.bilibili.tv") {
+		switch (document.location.pathname.toString().toLowerCase().split("/")[2]) {
```
Comment (ImRodry) [inline websites/B/bilibilitv/presence.ts:53]: you're stringifying a string?
```suggestion
		switch (document.location.pathname.toLowerCase().split("/")[2]) {
```

File: websites/B/bilibilitv/presence.ts [hunk 2]
```diff
@@ -1,97 +1,164 @@
...
+			case "video": {
+				presenceData.details = strings.watchingVid;
+				presenceData.state = title;
+				presenceData.startTimestamp = browsingStamp;
```
Comment (ImRodry) [inline websites/B/bilibilitv/presence.ts:57]: this has already been set, no need to set it twice

File: websites/B/bilibilitv/presence.ts [hunk 3]
```diff
@@ -1,97 +1,164 @@
...
+						.querySelector("span.series-text.active")
+						.textContent.replace(/ *\([^)]*\) */g, "")} | ${strings.episode} ${
+						title.match(/\d+/g)[0]
+					}`;
+				} else ep = `${strings.episode} ${title.match(/\d+/g)[0]}`;
+				presenceData.details = `${
+					document.querySelector("div.video-info__title-wrap > h1 > a")
+						.textContent
+				}`;
+				presenceData.state = `${ep}`;
```
Comment (ImRodry) [inline websites/B/bilibilitv/presence.ts:94]: no need to stringify strings
```suggestion
				presenceData.state = ep;
```

File: websites/B/bilibilitv/presence.ts [hunk 4]
```diff
@@ -1,97 +1,164 @@
...
+				presenceData.smallImageKey = playing ? "pause" : "play";
```
Comment (ImRodry) [inline websites/B/bilibilitv/presence.ts:95]: if playing is true the key is pause? Sure?

File: websites/B/bilibilitv/presence.ts [hunk 5]
```diff
@@ -1,97 +1,164 @@
...
+			default: {
+				presenceData.details = strings.viewHome;
+				presenceData.smallImageKey = "reading";
+				presenceData.smallImageText = strings.browsing;
+				break;
+			}
+		}
+		// Studio
+	} else if (document.location.hostname === "studio.bilibili.tv") {
+		if (document.location.pathname.toString().toLowerCase().split("/")[1]) {
```
Comment (ImRodry) [inline websites/B/bilibilitv/presence.ts:154]:
```suggestion
		if (document.location.pathname.toLowerCase().split("/")[1]) {
```

File: websites/B/bilibilitv/presence.ts [hunk 6]
```diff
@@ -1,97 +1,164 @@
...
+			presenceData.details =
+				document.querySelector(".is-active")?.textContent ?? "?";
```
Comment (ImRodry) [inline websites/B/bilibilitv/presence.ts:156]: you're setting the details to "?" if that query selector is null?

**Review by ImRodry — CHANGES_REQUESTED** [review]

File: websites/B/bilibilitv/presence.ts (hunk 4 above)
Comment (ImRodry) [reply websites/B/bilibilitv/presence.ts:92]: Don't resolve conversations without addressing them in some form.

File: websites/B/bilibilitv/presence.ts [hunk 7]
```diff
@@ -1,97 +1,164 @@
...
-	};
+	browsingStamp = Math.floor(Date.now() / 1000);
 
-presence.on("UpdateData", async () => {
-	const presenceData: PresenceData = {
-			largeImageKey: "icon",
-			details: "Idle"
+async function getStrings() {
+	return presence.getStrings(
+		{
```
[the remark this answers is not in the document]
Comment (Faelayis) [reply websites/B/bilibilitv/presence.ts:8]: Lint will not pass

File: websites/B/bilibilitv/presence.ts [hunk 8]
```diff
@@ -1,97 +1,161 @@
...
+				presenceData.smallImageKey = "reading";
+				presenceData.smallImageText = strings.browsing;
+				break;
+			}
+			case "space": {
+				presenceData.details = strings.viewChannel.replace(":", "");
+				presenceData.state = `${
+					document.querySelector(".space-info__name")?.textContent ??
+					strings.searchSomething
+				}`;
```
Comment (ImRodry) [inline websites/B/bilibilitv/presence.ts:129]:
```suggestion
				presenceData.state =
					document.querySelector(".space-info__name")?.textContent ??
					strings.searchSomething;
```

File: websites/B/bilibilitv/presence.ts [hunk 9]
```diff
@@ -1,97 +1,161 @@
...
+					ep = `${document
+						.querySelector("span.series-text.active")
+						.textContent.replace(/ *\([^)]*\) */g, "")} | ${strings.episode} ${
+						title.match(/\d+/g)[0]
+					}`;
+				} else ep = `${strings.episode} ${title.match(/\d+/g)[0]}`;
+				presenceData.details = `${
+					document.querySelector("div.video-info__title-wrap > h1 > a")
+						.textContent
+				}`;
```
Comment (ImRodry) [inline websites/B/bilibilitv/presence.ts:90]:
```suggestion
				presenceData.details =
					document.querySelector("div.video-info__title-wrap > h1 > a")
						.textContent;
```

**Review by ImRodry — CHANGES_REQUESTED** [review]
Seems like I missed two

**Review by ImRodry — DISMISSED** [review]

**Comment by ImRodry** [comment]
Make sure to fix the CI issue

File: websites/B/bilibilitv/presence.ts [hunk 10]
```diff
@@ -1,97 +1,159 @@
...
+				presenceData.details = document.querySelector(
+					"div.video-info__title-wrap > h1 > a"
+				).textContent;
+				presenceData.state = ep;
+				presenceData.smallImageKey = playing ? "play" : "pause";
+				presenceData.smallImageText = playing ? strings.play : strings.pause;
+				presenceData.largeImageKey = document
+					.querySelector<HTMLImageElement>('meta[name="og:image"]')
+					.getAttribute("content")
+					.split("@")[0];
```
Comment (Slowlife01) [inline websites/B/bilibilitv/presence.ts:96]:
```suggestion
					.querySelector<HTMLMetaElement>('meta[name="og:image"]')
					.content.split("@")[0];
```

File: websites/B/bilibilitv/presence.ts [hunk 11]
```diff
@@ -1,97 +1,159 @@
...
+				presenceData.buttons = [
+					{
+						label: strings.buttonViewEpisode,
+						url: document
+							.querySelector('meta[name="og:url"]')
+							.getAttribute("content")
```
Comment (Slowlife01) [inline websites/B/bilibilitv/presence.ts:102]: Can't you use just location.href?

File: websites/B/bilibilitv/presence.ts [hunk 12]
```diff
@@ -1,97 +1,159 @@
...
+		switch (document.location.pathname.toLowerCase().split("/")[2]) {
+			case "video": {
+				presenceData.details = strings.watchingVid;
+				presenceData.state = title;
+				presenceData.smallImageKey = playing ? "play" : "pause";
+				presenceData.smallImageText = playing ? strings.play : strings.pause;
+				presenceData.largeImageKey = document
+					.querySelector<HTMLImageElement>('meta[name="og:image"]')
+					.getAttribute("content")
+					.split("@")[0];
```
Comment (Slowlife01) [inline websites/B/bilibilitv/presence.ts:60]:
```suggestion
				presenceData.largeImageKey = document
					.querySelector<HTMLMetaElement>('meta[name="og:image"]')
					.content.split("@")[0];
```

**Review by ImRodry — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**